lib/src/accordion/accordion.component.ts

```diff
@@ -0,0 +1,175 @@
+
+    constructor(private changeDetector: ChangeDetectorRef) {}
+
+    //helper functions
+    constructIds() {
+        this.idList = this.list.map(() => randomId("accordion-"));
+    }
+
+    /** Constructs the `className` to be used in accordion wrapper */
+    constructClassName(): void {
+        let cn: string = "custom-accordion";
+        cn += this.className ? ` ${this.className}` : "";
+        cn += this.alternative ? " alternative-accordion" : "";
+        this.accordionClassName = cn;
+    }
+
+    /** Constructs the `className` to be used in accordion items */
+    constructItemClassName(): void {
+        let cn: string = "accordion-item";
+        cn += " " + (this.iconPosition ? this.iconPosition : "left");
+        cn += " " + (this.iconRotation ? this.iconRotation : "deg-180");
+        cn += this.customIconExpanded ? " transform" : "";
+        this.itemClassName = cn;
+    }
+
+    expandOrCollapseSection(itemIndex: number): void {
+        const updatedHeightList: Array<number> = Array(this.list?.length).fill(0);
+        updatedHeightList[itemIndex] = this.heightList[itemIndex]
+            ? 0
+            : this.accordionItemRefs?.toArray()[itemIndex]?.nativeElement?.scrollHeight;
+        this.heightList = updatedHeightList;
+    }
+
+    toggle(index: number): void {
+        if (this.active === index) {
+            // Section already expanded
+            this.expandOrCollapseSection(index);
+        } else {
+            if (this.active !== null) {
+                // Another section is already expanded
+                this.expandOrCollapseSection(this.active);
+            }
+            this.expandOrCollapseSection(index);
+        }
+        this.active = this.active === index ? null : index;
+    }
+
+    contentIsOfTypeArray(content: AccordionContentType): boolean {
+        return content instanceof Array;
+    }
+
+    contentTypeElementRef(content: AccordionContentType): boolean {
+        return typeof content === "string";
+    }
+
+    // events ----------------------------------------------
+    /**
+     * Handles accordion item click event
+     * @param e MouseEvent
+     * @param index list index
+     */
+    onToggle(e: KeyboardEvent | MouseEvent): void {
+        const index: number = Number((e.currentTarget as HTMLElement).getAttribute("data-id"));
+        if (e.type === "keydown") {
+            const key: string = (e as KeyboardEvent).key;
+            if ([" ", "space", "enter"].indexOf(key.toLowerCase()) !== -1) {
+                this.toggle(index);
+                e.preventDefault();
+            }
+        } else {
+            this.toggle(index);
+        }
+    }
+
+    ngOnInit() {
+        this.constructIds();
+        this.constructClassName();
+        this.constructItemClassName();
+        this.heightList = Array(this.list?.length).fill(0);
+    }
+
+    ngAfterViewInit() {
+        this.toggle(this.activeIndex);
+        this.accordionItemRefs.changes.subscribe(r => {
```

Comment:
if you're not using `r`, I believe you can omit it
```suggestion
        this.accordionItemRefs.changes.subscribe(() => {
```